extraction/image_extractor.py

```diff
@@ -0,0 +1,40 @@
+"""This script checks whether DOM has image tag or not and
+creates and returns the Image object"""
+
+from extraction.data_models.image import Image
+from extraction.interface_content_extractor import IContentExtractor
+
+
+class ImageExtractor(IContentExtractor):
+    """This class inherits IContentExtractor to extract Images"""
+
+    def get_size(self, node):
+        """Returns the size of the image"""
+
+        if node.has_attr('width') and node.has_attr('height'):
+            return node['width'], node['height']
+        return 0, 0
+
+    def get_image_attributes(self, node, image):
+        """This method extracts all attributes for the image"""
+
+        image.img_url = node['src']
+        if node.has_attr('title'):
+            image.img_title = node['title']
+        image.img_height, image.img_width = self.get_size(node)
+        image.is_gif = image.img_url.endswith('.gif')
+
+    def validate_and_extract(self, node):
+        if node.name == 'img' and node.has_attr('src'):
+            image_content = Image()
+            self.get_image_attributes(node, image_content)
+            return image_content
+        if node.name == 'figure':
+            img_tag = node.find('img')
+            if img_tag and img_tag.has_attr('src'):
+                image_content = Image()
```

Comment:
Its better to parameterised constructor with agruments as all element set in get_image_attributes

Reply:
Update `get_image_attributes` to `create_image`. `create_image` can return Image(with, all, the, arguments)

Reply:
Updated accordingly! Thanks for the suggestions.